atc/api/pipelineserver/reject_archived_handler_factory.go

```diff
@@ -0,0 +1,63 @@
+package pipelineserver
+
+import (
+	"net/http"
+
+	"github.com/concourse/concourse/atc/db"
+)
+
+type RejectArchivedHandlerFactory struct {
+	teamFactory db.TeamFactory
+}
+
+func NewRejectArchivedHandlerFactory(factory db.TeamFactory) RejectArchivedHandlerFactory {
+	return RejectArchivedHandlerFactory{
+		teamFactory: factory,
+	}
+}
+
+func (f RejectArchivedHandlerFactory) RejectArchived(handler http.Handler) http.Handler {
+	return RejectArchivedHandler{
+		teamFactory:     f.teamFactory,
+		delegateHandler: handler,
+	}
+}
+
+type RejectArchivedHandler struct {
+	teamFactory     db.TeamFactory
+	delegateHandler http.Handler
+}
+
+func (ra RejectArchivedHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	teamName := r.FormValue(":team_name")
+	pipelineName := r.FormValue(":pipeline_name")
+
+	team, found, err := ra.teamFactory.FindTeam(teamName)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
nit: I think this error scenario is not covered :thinking:  aside from ` 	Context("when a team is not found", func() {`, we'd then have a "when finding a team fails" I guess